arangod/RocksDBEngine/RocksDBCollectionMeta.cpp

```diff
@@ -0,0 +1,419 @@
+      continue;
+    }
+    
+    key.constructIndexEstimateValue(idx->objectId());
+    s = db->Get(ro, cf, key.string(), &value);
+    if (!s.ok() && !s.IsNotFound()) {
+      return rocksutils::convertStatus(s);
+    } else if (s.IsNotFound()) { // expected with nosync recovery tests
+      LOG_TOPIC(WARN, Logger::ROCKSDB) << "recalculating index estimate for index "
+      << "type '" << idx->typeName() << "' with id '" << idx->id() << "'";
+      idx->recalculateEstimates();
+      continue;
+    }
+    
+    StringRef estimateInput(value.data() + sizeof(uint64_t),
+                            value.size() - sizeof(uint64_t));
+    
+    uint64_t committedSeq = rocksutils::uint64FromPersistent(value.data());
+    if (RocksDBCuckooIndexEstimator<uint64_t>::isFormatSupported(estimateInput)) {
+      auto est = std::make_unique<RocksDBCuckooIndexEstimator<uint64_t>>(committedSeq, estimateInput);
+      LOG_TOPIC(ERR, Logger::ENGINES)
+      << "found index estimator for objectId '" << idx->objectId()
+      << "' committed seqNr '" << committedSeq << "' with estimate "
+      << est->computeEstimate();
+      
+      idx->setEstimator(std::move(est));
+    }
+  }
+  
+  return Result();
+}
+
+/// @brief load collection
+/*static*/ RocksDBCollectionMeta::DocCount
+    RocksDBCollectionMeta::loadCollectionCount(rocksdb::DB* db, uint64_t objectId) {
+      
+  auto cf = RocksDBColumnFamily::definitions();
+  rocksdb::ReadOptions ro;
+  ro.fill_cache = false;
+  
+  RocksDBKey key;
+  key.constructCounterValue(objectId);
+  
+  rocksdb::PinnableSlice value;
+  rocksdb::Status s = db->Get(ro, cf, key.string(), &value);
+  if (s.ok()) {
+    VPackSlice countSlice = RocksDBValue::data(value);
+    return RocksDBCollectionMeta::DocCount(countSlice);
+  }
+  return DocCount(0,0,0,0);
+}
+
+/// @brief remove collection metadata
+/*static*/ Result RocksDBCollectionMeta::deleteCollectionMeta(rocksdb::DB* db, uint64_t objectId) {
+  
+
+  rocksdb::ColumnFamilyHandle* const cf = RocksDBColumnFamily::definitions();
+  rocksdb::WriteOptions wo;
+  wo.sync = true;
+  
+  
+  // Step 1. delete the document count
+  RocksDBKey key;
+  key.constructCounterValue(objectId);
+  rocksdb::Status s =db->Delete(wo, cf, key.string());
+  if (!s.ok()) {
+    return rocksutils::convertStatus(s);
+  }
+  
+  key.constructKeyGeneratorValue(objectId);
+  s = db->Delete(wo, cf, key.string());
+  if (!s.ok() && !s.IsNotFound()) {
+    return rocksutils::convertStatus(s);
+  }
+  
+  return Result();
+}
+
+/// @brief remove collection index estimate
+/*static*/ Result RocksDBCollectionMeta::deleteIndexEstimate(rocksdb::DB* db, uint64_t objectId) {
+  
+  rocksdb::ColumnFamilyHandle* const cf = RocksDBColumnFamily::definitions();
+  rocksdb::WriteOptions wo;
+  wo.sync = true;
+  
+  RocksDBKey key;
+  key.constructIndexEstimateValue(objectId);
+  rocksdb::Status s = db->Delete(wo, cf, key.string());
+  if (!s.ok() && !s.IsNotFound()) {
+    return rocksutils::convertStatus(s);
```

Comment:
shouldn't we just go on deleting the estimates for other remaining indexes of the collection? otherwise even more "garbage" will remain.